### PR TITLE
Switch to using catalog discovery client

### PR DESCRIPTION
**Pull request by timyates**

### CODE REVIEW

File: aws-distributed-configuration/build.gradle.kts [hunk 1]
```diff
@@ -4,6 +4,6 @@ plugins {
 
 dependencies {
     api(project(":aws-common"))
-    api("io.micronaut.discovery:micronaut-discovery-client")
+    api(mn.micronaut.discovery)
```
Comment (sdelamo) [inline aws-distributed-configuration/build.gradle.kts:7]: I don't understand. 

should not be `artifactId` replacing `-` with `.` and preffixing with `mn.` ?

Comment (sdelamo) [reply aws-distributed-configuration/build.gradle.kts:7]: @melix @timyates  ?

**Review by sdelamo — APPROVED** [review]

File: aws-distributed-configuration/build.gradle.kts (hunk 1 above)
Comment (melix) [reply aws-distributed-configuration/build.gradle.kts:7]: It only depends on how it was written in the catalog in the beginning

Comment (timyates) [reply aws-distributed-configuration/build.gradle.kts:7]: And it's been "discovery" for over a year

https://github.com/micronaut-projects/micronaut-core/blob/3.6.x/gradle/libs.versions.toml#L282

**Pull request merged**